core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -391,7 +392,8 @@ private void notifyConnect(SessionProtocol desiredProtocol, PoolKey key, Future<
                 final SessionProtocol protocol = getProtocolIfHealthy(channel);
                 if (closed || protocol == null) {
                     channel.close();
-                    promise.completeExceptionally(UnprocessedRequestException.get());
+                    promise.completeExceptionally(
+                            new UnprocessedRequestException(ClosedSessionException.get()));
```

Comment:
There are two more `completeExceptionally()` methods below in this method you need to take care of.

Reply:
Thanks! I wrapped below exception with UnprocessedRequestException.  :-)